x-pack/metricbeat/module/meraki/wireless_device_channel_utilization/wireless_device_channel_utilization.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/meraki"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	meraki_api "github.com/meraki/dashboard-api-go/v3/sdk"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet(meraki.ModuleName, "wireless_device_channel_utilization", New)
+}
+
+type config struct {
+	BaseURL       string   `config:"apiBaseURL"`
+	ApiKey        string   `config:"apiKey"`
+	DebugMode     string   `config:"apiDebugMode"`
+	Organizations []string `config:"organizations"`
+	// todo: device filtering?
+}
+
+func defaultConfig() *config {
+	return &config{
+		BaseURL:   "https://api.meraki.com",
+		DebugMode: "false",
+	}
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	logger        *logp.Logger
+	client        *meraki_api.Client
+	organizations []string
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The meraki wireless_device_channel_utilization metricset is beta.")
+
+	logger := logp.NewLogger(base.FullyQualifiedName())
+
+	config := defaultConfig()
+	if err := base.Module().UnpackConfig(config); err != nil {
+		return nil, err
+	}
+
+	logger.Debugf("loaded config: %v", config)
+	client, err := meraki_api.NewClientWithOptions(config.BaseURL, config.ApiKey, config.DebugMode, "Metricbeat Elastic")
+	if err != nil {
+		logger.Error("creating Meraki dashboard API client failed: %w", err)
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+		logger:        logger,
+		client:        client,
+		organizations: config.Organizations,
+	}, nil
+}
+
+// Fetch method implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
+	for _, org := range m.organizations {
+
+		devices, err := meraki.GetDevices(m.client, org)
+		if err != nil {
+			return err
+		}
+
+		res, err := m.client.Devices.GetOrganizationWirelessDevicesChannelUtilizationByDevice(org, &meraki_api.GetOrganizationWirelessDevicesChannelUtilizationByDeviceQueryParams{})
```

Comment:
by default this returns a weeks worth of data

same questions here as on `GetNetworkNetworkHealthChannelUtilization` - i.e. should we just use a time period equal to the collection period? and if so what resolution makes sense (here we have more options - 300, 600, 3600, 7200, 14400, 21600, default 3600).

Reply:
I like 600, but make it configurable in the modules.d/meraki.yml

Reply:
i wonder if we should try to increase the resolution to be at least as large as the interval so that we only have one metric data point per collection. 

we could acheive this by specificying a resolution of 600 in both cases, and putting a maximum collection interval for the metricset of 10 minutes. 

if we don't want to put that restriction on the collection interval, it's probably best to report the datapoint for each interval as a seperate metric (with the timestamp equal to the end of the metric interval period). if we are reporting metric docs with @timestamp, but the actual time period the metric was calculated for is different, it makes using the metrics in the backend (i.e. kibana, lens) realy painful. 

what do you think?

Reply:
It makes sense that the time interval would need to be the consistent between the meraki API call time interval and the module interval.   However, I worry slightly about the other metrics that might be used for alerting, where devices can go into failed state, they may want more frequent time checks, and I am not sure a 10 minute time interval would be appropriate.  Maybe we  should get input from a meraki user and see what they think the default limit should, be, especially given there is meraki api throttling of 10 calls per second on the API calls. Then here is may question, if we can just pull from the modules config the module.period, extract it and use that as the time interval so they are both consistent. So if they set module period time at 5 minutes, we would just harvest that value and set accordingly on the API call ??  In that case would that be sufficient or do we still need the metric for end of time interval period, since it would not be static?

Reply:
i was suggesting a **max** interval of 10 minutes just to make it easier on us to report, since then we can only ever get one bucket per collection from the API. if we run collections every one minute, we still use a 600s resolution on this metric, and (hopefully) we just get one bucket with 60s worth of data in it.

maybe i'm overthinking it, and we should just process each bucket regardless of collection interval, and potentially report metrics with multiple timestamps for each colleciton period.

@agithomas do you have any thoughts about this?

Reply:
So upon thinking on this further, I think I understand better what your saying, the MB capture interval can be whatever an user wants to set, however we just always capture at a static value of 10min and it does not or should not be configurable for dashboard purposes ...  I am fine with that, I was thinking it always had to be the data in-between time intervals ... but I am fine with the 10 minute constant.

Reply:
Unless we hear otherwise, perhaps we just leave it the default of 3600, only because I have no clue why 10 minutes vs 1hr would be better ... ?? I am not a dashboard person. :)  @agithomas  do you have any further input? api defaults to 3600 seconds (1hr) ... we can modify to 5 minute, or 10 minute intervals, if preferred.

Reply:
Leaving at default settings.